src/openassetio-core/include/openassetio/managerApi/ManagerInterface.hpp

```diff
@@ -938,6 +937,77 @@ class OPENASSETIO_CORE_EXPORT ManagerInterface {
                             const ExistsSuccessCallback& successCallback,
                             const BatchElementErrorCallback& errorCallback);
 
+  /**
+   * Callback signature used for a successful entity trait set query.
+   */
+  using EntityTraitsSuccessCallback = std::function<void(std::size_t, trait::TraitSet)>;
+
+  /**
+   * Provides the host with the @ref trait_set of one or more @ref
+   * entity "entities".
+   *
+   * For example, a host may use this to validate that a user-provided
+   * entity reference is appropriate for an operation.
+   *
+   * The trait set returned (via callback) for each @ref
+   * entity_reference should vary according to the @p entityTraitsAccess
+   * access mode.
+   *
+   * If @ref access.EntityTraitsAccess.kRead "kRead" is given, respond
+   * with an exhaustive trait set for the entity. If an entity does not
+   * exist, then call the error callback using the @ref
+   * errors.BatchElementError.ErrorCode.kEntityResolutionError
+   * "kEntityResolutionError" code.
+   *
+   * If @ref access.EntityTraitsAccess.kWrite "kWrite" is given, respond
+   * with the minimal trait set required to publish to the entity
+   * reference. If an entity is read-only, then call the error callback
+   * using the @ref
+   * errors.BatchElementError.ErrorCode.kEntityAccessError
+   * "kEntityAccessError" code.
+   *
+   * The trait set should include all traits that an entity possesses,
+   * not just those with `resolve`able properties, so that the kind of
+   * entity can be determined by the host.
```

Comment:
[Consider] Reads a bit strange coming out of the write section where we say return a minimal traitset. Assuming you're trying to make this point about both read and write, perhaps?

```suggestion
   * When populating the `entityTraits` return, you should include all
   * relevant traits for the access mode, not just those with `resolve`able properties.
```
The "so that the kind of entity can be determined by the host" seems only correct when applied to read, i'd just omit it.

Reply:
Did as suggested in d4ec12ca, but then paired with Tom on a rewrite in 188db754 and 1dfd5f05